src/wrap.rs

```diff
@@ -25,8 +25,13 @@ impl<'a> SpansWrapper<'a> {
             if let Some(s) = self.pending.take() {
                 span = Some(s);
             } else if self.index < self.spans.len() {
-                span = Some(self.spans.get(self.index).cloned().unwrap());
-                self.index += 1;
+                match self.spans.get(self.index).cloned() {
```

Comment:
Thanks @5ylar for the patch! Maybe I'm missing something, could you clarify in which scenario would the unwrap panic if we've checked that `self.index` is within bound? A test case would be helpful.